mne/beamformer/_lcmv.py

```diff
@@ -24,6 +24,16 @@
 from ..channels.channels import _contains_ch_type
 
 
+def _deprecate_picks(data_obj, picks):
+    if picks is not None:
+        warn('Specifying picks is deprecated and will be removed in 0.16. '
+             'Specifying the selection of channels in info and setting picks '
+             'to None will remove this warning.',
+             DeprecationWarning)
+
+        data_obj.info = pick_info(data_obj.info, picks)
```

Comment:
I agree that it's not optimal: the info here is passed to ``_setup_picks`` whose output is then passed to further functions which do the actual picking (e.g., forward model is modified in ``_prepare_beamformer_input`` and data in the ``apply_lcmv`` functions).  
Initially I wanted to integrate all this in one function, but the objects are handled differently in ``_lcmv_source_power`` and ``_apply_lcmv``. Ultimately, ``_lcmv_source_power`` should become an option of ``_apply_lcmv`` rather than being an own function, I guess, and then it's much more straight forward to put all of the picking in one place.